Sources/BraveWallet/Crypto/Asset Details/AssetDetailView.swift

```diff
@@ -36,173 +36,226 @@ struct AssetDetailView: View {
+  
+  private var emptyTransactionState: some View {
+    VStack(spacing: 10) {
+      Image("transaction-empty", bundle: .module)
+        .aspectRatio(contentMode: .fit)
+      Text(Strings.Wallet.noTransactions)
+        .font(.headline)
+        .foregroundColor(Color(WalletV2Design.textPrimary))
+      Text(Strings.Wallet.activityPageEmptyDescription)
+        .font(.footnote)
+        .foregroundColor(Color(WalletV2Design.textSecondary))
+    }
+    .multilineTextAlignment(.center)
+    .padding(.vertical)
+  }
+  
+  @ViewBuilder private var tokenContentDrawer: some View {
+    VStack(spacing: 0) {
+      AssetDetailSegmentedControl(selected: $selectedContent)
+        .padding(.horizontal)
+      if selectedContent == .accounts {
+        accountsBalanceView
+          .padding(.horizontal)
+          .padding(.top, 16)
+      } else {
+        if assetDetailStore.transactionSections.isEmpty {
+          emptyTransactionState
+        } else {
+          TransactionsListView(
+            transactionSections: assetDetailStore.transactionSections,
+            query: $query,
+            showFilter: false,
+            filtersButtonTapped: {},
+            transactionTapped: { tx in
+              self.transactionDetails = assetDetailStore.transactionDetailsStore(for: tx)
+            }
+          )
         }
-        Spacer()
-        VStack(spacing: 10) {
-          let computed = assetDetailStore.currencyFormatter.string(from: NSNumber(value: BraveWallet.CoinMarket.abbreviateToBillion(input: coinMarket.marketCap))) ?? ""
-          Text("\(computed)B")
-            .font(.title3.weight(.semibold))
-          Text(Strings.Wallet.coinMarketMarketCap)
+      }
+    }
+  }
+  
+  private var assetDetailContentView: some View {
+    LazyVStack {
+      switch assetDetailStore.assetDetailType {
+      case .blockchainToken(_):
+        tokenContentDrawer
+          .padding(.bottom, 12)
+        
+        if selectedContent != .transactions || !assetDetailStore.transactionSections.isEmpty {
+          Text(Strings.Wallet.coinGeckoDisclaimer)
+            .multilineTextAlignment(.center)
             .font(.footnote)
+            .foregroundColor(Color(.secondaryBraveLabel))
+            .frame(maxWidth: .infinity)
+            .padding(.bottom, 12)
         }
+      case .coinMarket(let coinMarket):
+        coinMarketInfoView(coinMarket)
+          .padding(.bottom, 20)
       }
-      .foregroundColor(Color(.braveLabel))
-      .listRowBackground(Color(.secondaryBraveGroupedBackground))
-    } header: {
-      Text(Strings.Wallet.coinMarketInformation)
     }
+    .padding(.vertical)
+    .background(
+      ZStack {
+        Color(braveSystemName: .pageBackground) // bg behind rounded corners
+          .zIndex(0)
+        Color(braveSystemName: .containerBackground)
+          .roundedCorner(16, corners: [.topLeft, .topRight])
+          .shadow(color: .black.opacity(0.04), radius: 8, x: 0, y: -8)
+          .zIndex(1)
+      }
+    )
```

Comment:
Very minor bug but the background leaks through when scrolling. Any way to fix that? It's more noticable on iPads.

![Simulator Screenshot - iPhone 15 - 2024-01-02 at 11 13 10](https://github.com/brave/brave-ios/assets/5314553/00c528a0-e32d-4731-ba39-cb0fa141d088) | ![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2024-01-02 at 11 16 41](https://github.com/brave/brave-ios/assets/5314553/0e232500-24f4-4ac5-89e9-623399539f23)
--|--

Reply:
Actually double checking design we aren't using the tray style in asset detail, it's a solid background:
<img width="347" alt="Asset Detail" src="https://github.com/brave/brave-ios/assets/5314553/bf2395f8-4755-4903-b3a4-3c95bb5b52ad">

Reply:
removed the tray style in asset details and coin market details. confirmed with @kleantzogu

Reply:
`AssetDetailHeaderView` still has `pageBackground` instead of `containerBackground` background colour.